### PR TITLE
52201-fss api and configuation

### PR DESCRIPTION
**Pull request by Vishal14583**

### CODE REVIEW

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Helpers/FssApiClient.cs [hunk 1]
```diff
@@ -1,26 +1,123 @@
...
+
+            return await _httpClient.SendAsync(httpRequestMessage, CancellationToken.None);
+        }
+
+        public async Task<HttpResponseMessage> UploadFileBlockAsync(string uri, byte[] blockBytes, byte[] md5Hash, string accessToken, string mimeTypeHeader = "application/octet-stream")
+        {
+            using var httpRequestMessage = new HttpRequestMessage(HttpMethod.Put, uri)
+            { Content = (blockBytes == null) ? null : new ByteArrayContent(blockBytes) };
+
+            httpRequestMessage.Content.Headers.ContentType = new MediaTypeHeaderValue("application/octet-stream");
```
Comment (harshal11869) [inline UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Helpers/FssApiClient.cs:54]: Can we add not null condition before this as httpRequestMessage.Content can be null

**Review by harshal11869 — CHANGES_REQUESTED** [review]

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Helpers/FssApiClient.cs [hunk 2]
```diff
@@ -1,26 +1,123 @@
...
     {
         private readonly HttpClient _httpClient;
 
-        public FssApiClient(IHttpClientFactory httpClientFactory)
+        public FssApiClient(IHttpClientFactory httpClientFactory) => _httpClient = httpClientFactory.CreateClient();
+
+        public async Task<HttpResponseMessage> CreateBatchAsync(string uri, string requestBody, string authToken)
         {
-            _httpClient = httpClientFactory.CreateClient();
+            HttpContent content = null;
```
[the remark this answers is not in the document]
Comment (Vishal14583) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Helpers/FssApiClient.cs]: Resolved

File: UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Helpers/FssApiClient.cs (hunk 1 above)
Comment (Vishal14583) [reply UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Helpers/FssApiClient.cs]: Resolved

**Review by harshal11869 — APPROVED** [review]
LGTM

**Review by ajasonca — CHANGES_REQUESTED** [review]
Some questions/changes to think about.

**Review by ajasonca — APPROVED** [review]
Looking good

**Pull request merged**